src/content/docs/en/recipes/making-toolbar-apps.mdx

```diff
@@ -0,0 +1,229 @@
+---
+title: Creating a dev toolbar app
+description: Learn how to create a dev toolbar app for your site.
+type: recipe
+---
+
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import { FileTree } from '@astrojs/starlight/components';
+
+Astro includes a [development toolbar](/en/guides/dev-toolbar/) that you can use to inspect your site, check for accessibility and performance issues, and more. 
+
+In this recipe, you'll learn how to create a dev toolbar app that you can use to extend the functionality of the dev toolbar.
+
+:::tip 
+Just wanna get started quickly and start working on your app? Jump start your app by creating a new Astro project with the `toolbar-app` template.
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+    ```shell
+    npm create astro@latest -- --template toolbar-app
+    ```
+  </Fragment>
+  <Fragment slot="pnpm">
+    ```shell
+    pnpm create astro -- --template toolbar-app
+    ```
+  </Fragment>
+  <Fragment slot="yarn">
+    ```shell
+    yarn create astro -- --template toolbar-app
+    ```
+  </Fragment>
+</PackageManagerTabs>
+
+Alternatively, keep on reading to learn how to build an app from scratch.
+:::
+
+## Creating the integration
+
+Dev toolbar apps can only be added by [Astro Integrations](/en/guides/integrations-guide/). To create an integration, create a file anywhere in your project exporting an object with the integration name and hooks. 
+
+The `astro:config:setup` hook has access to the `addDevToolbarApp` function, which can be used to add dev toolbar apps. The `addDevToolbarApp` function takes an object with the id, name, icon and finally entrypoint of the dev toolbar app. The entrypoint is the path to the file that exports the dev toolbar app.
+
+```ts title="my-integration.ts"
+import { fileURLToPath } from 'node:url';
+import { dirname, join } from 'node:path';
+import type { AstroIntegration } from 'astro';
+
+export default {
+  name: 'my-astro-integration',
+  hooks: {
+    'astro:config:setup': ({ addDevToolbarApp }) => {
+      const __filename = fileURLToPath(import.meta.url);
+      const __dirname = dirname(__filename);
+  
+      addDevToolbarApp({
+        id: "my-toolbar-app",
+        name: "My Toolbar App",
+        icon: "🚀",
+        entrypoint: join(__dirname, "./app.ts"),
```

Comment:
All the path code doesn't look great. I don't know why we left all of these entrypoints as always being string. Can we at least clear this up a little by reducing to `fileURLToPath(new URL('./app.ts', import.meta.url))` ?